serde2/serde2_macros/src/lib.rs

```diff
@@ -842,18 +850,50 @@ fn declare_map_field_deserializer(
     ]
 }
 
+
+fn default_value(field: &ast::StructField) -> bool {
+    field.node.attrs.iter()
+        .any(|sa|
+             match &sa.node.value.node {
+                 &ast::MetaItem_::MetaList(ref n, ref vals) => {
+                     if n == &"serde" {
+                         vals.iter()
+                             .map(|mi|
+                                  match &mi.node {
```

Comment:
These `match`es could be refactored into `if let`, that would remove 2 levels of indentation.